vidlistener/listener.go

```diff
@@ -22,21 +22,21 @@ type VidListener struct {
 	RtmpServer *joy4rtmp.Server
 }
 
-//HandleRTMPPublish takes 3 parameters - makeStreamID, gotStream, and endStream.
-//makeStreamID is called when the stream starts. It should return a streamID from the requestURL.
-//gotStream is called when the stream starts.  It gives you access to the stream.
-//endStream is called when the stream ends.  It gives you access to the stream.
+// HandleRTMPPublish takes 3 parameters - makeStreamID, gotStream, and endStream.
+// makeStreamID is called when the stream starts. It should return a streamID from the requestURL.
+// gotStream is called when the stream starts.  It gives you access to the stream.
+// endStream is called when the stream ends.  It gives you access to the stream.
 func (self *VidListener) HandleRTMPPublish(
-	makeStreamID func(url *url.URL) (strmID stream.AppData),
+	makeStreamID func(url *url.URL) (strmID stream.AppData, err error),
 	gotStream func(url *url.URL, rtmpStrm stream.RTMPVideoStream) error,
 	endStream func(url *url.URL, rtmpStrm stream.RTMPVideoStream) error) {
 
 	if self.RtmpServer != nil {
 		self.RtmpServer.HandlePublish = func(conn *joy4rtmp.Conn) {
 			glog.V(2).Infof("RTMP server got upstream: %v", conn.URL)
 
-			strmID := makeStreamID(conn.URL)
-			if strmID == nil || strmID.StreamID() == "" {
```

Comment:
Where is this `makeStreamID()` actually implemented? In go-livepeer? I wonder if changing this condition here won't bring any regression.

Reply:
Yep, in `go-livepeer`. I think that we can probably keep the original check as well though if we'd like to be safe